timesketch/views/auth.py

```diff
@@ -156,6 +167,88 @@ def logout():
     return redirect(url_for('user_views.login'))
 
 
+@auth_views.route('/login/api_callback/', methods=['GET'])
+def validate_api_token():
+    """Handler for logging in using an authenticated session for the API.
+
+    Returns:
+        A simple page indicating the user is authenticated.
+    """
+    try:
+        token = oauth2.rfc6749.tokens.get_token_from_header(request)
+    except AttributeError:
+        token = None
+
+    if not token:
+        return abort(
+            HTTP_STATUS_CODE_UNAUTHORIZED, 'Request not authenticated.')
+
+    client_id = current_app.config.get('GOOGLE_OIDC_API_CLIENT_ID')
+    if not client_id:
+        return abort(
+            HTTP_STATUS_CODE_BAD_REQUEST,
+            'No OIDC API client ID defined in the configuration file.')
+
+    token_check_url = '{0:s}?access_token={1:s}'.format(TOKEN_URI, token)
```

Comment:
Please add a comment with the url to the docs you are following here
https://www.oauth.com/oauth2-servers/signing-in-with-google/verifying-the-user-info/

Reply:
also, add a comment explining what is happening here. Just easier when we need to pick this up a year from now :)

Reply:
done